File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/13-RomanToInteger/13-RomanToInteger.cpp
// Last updated: 8/4/2026, 7:26:19 PM
class Solution {
public:
    int romanToInt(string s) {
        int n = s.length();
        int i = 0;
        int ans = 0;
        while(i < n){
            if(i+1 < n && helper(s[i+1]) > helper(s[i])){
                ans -= helper(s[i]);
            }
            else ans += helper(s[i]);
            i++;
        }

        return ans;
    }

    int helper(char c){
        if(c == 'I') return 1;
        else if(c == 'V') return 5;
        else if(c == 'X') return 10;
        else if(c == 'L') return 50;
        else if(c == 'C') return 100;
        else if(c == 'D') return 500;
        else return 1000;

    }
};

File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/13-RomanToInteger/13-RomanToInteger.java
// Last updated: 8/4/2026, 7:26:27 PM
class Solution {
    public int romanToInt(String s) {
        int num = 0 ; int ans = 0;

        for(int i = 0 ; i < s.length() ; i++){
            num = helper(s.charAt(i));

            if(i+1 < s.length() && num < helper(s.charAt(i+1))){
                ans = ans - num;
            }
            else{
                ans = ans + num;
            }
        }
        return ans;
    }
    public int helper(Character c){
        if(c == 'I') return 1;
        if(c == 'V') return 5;
        if(c == 'X') return 10;
        if(c == 'L') return 50;
        if(c == 'C') return 100;
        if(c == 'D') return 500;
        if(c == 'M') return 1000;
        else{
            return 0;
        }
    }
}